### PR TITLE
Date bug

### LINKED ISSUE #42: Fix date bug
## Task Description

- Currently regex can't find the date for some court cases

## User Stories

- As a developer, I need to get rid of any bugs in my code, so that it all runs smoothly

## Relevant Files [If Available] 

- extract.py

### PR DESCRIPTION
**Pull request by luciegamby-sigmalabs**
Added a lot more regex but then also an 'out' so we don't have to find regex patterns forever.

Closes #42

### CODE REVIEW

**Comment by luciegamby-sigmalabs** [comment]
Gonna write a strongly worded letter to the courtroom scribes after this and ask them to standardise their transcripts 😓

File: pipeline/extract.py [hunk 1]
```diff
@@ -123,21 +123,41 @@ def parse_pdf(court_case: dict):
...
+            judge = re.search(r"(?<=Before  : \n \n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(?<=Before : \n \n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(?<=Before:\n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(?<=BEFORE:\n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(THE HONOURABLE.*)", first_page)
+
```
Comment (ayeshaa63) [inline pipeline/extract.py:138]: Could you not put all your regex expressions together and just separate with a `|` ?

Comment (luciegamby-sigmalabs) [reply pipeline/extract.py:138]: I'll look into it a bit more! I tried it at first and I kept getting an error where the 'positive lookahead' part of my regex was of variable length. However, I'll keeping changing things around to see if i can work it out!

**Comment by luciegamby-sigmalabs** [comment]
I did try to address the regex but it wasn't working out. I used RegExr to help me with my expressions and see if they were valid when using '|' but when putting it into my code, it began failing. I'll try again if there's time in the future but for now, it is what it is.

File: pipeline/extract.py [hunk 2]
```diff
@@ -157,7 +158,7 @@ def parse_pdf(court_case: dict):
 
         court_case["conclusion"] = last_page
     except:
-        court_case = {}
```
Comment (MahinRahman8901) [inline pipeline/extract.py]: nice way to set the court case to be empty

File: pipeline/extract.py [hunk 3]
```diff
@@ -192,12 +193,9 @@ def extract_cases(pages: int) -> pd.DataFrame:
     for case_data in extracted_cases:
         download_pdfs(case_data)
         parse_pdf(case_data)
-        if not extracted_cases[j]:
-            extracted_cases.pop(j)
-        else:
-            j += 1
 
-    print(extracted_cases)
+    extracted_cases = list(filter(None, extracted_cases))
```
Comment (MahinRahman8901) [inline pipeline/extract.py:197]: love the way you simplified this

File: pipeline/extract.py [hunk 4]
```diff
@@ -123,21 +123,42 @@ def parse_pdf(court_case: dict):
...
+            case_no = re.search(
+                r"(LM.*)", first_page)
+        court_case["case_no"] = case_no.group(1).strip()
+
+        court_date = re.search(
+            r"(?<=Date: )(.*)", first_page)
+        if not court_date:
+            court_date = re.search(
+                r"(.* [0-9]* \w* [0-9]*)", first_page)
+
```
Comment (MahinRahman8901) [inline pipeline/extract.py:154]: Excellent use of regex here aswell

**Review by MahinRahman8901 — APPROVED** [review]

**Pull request merged**